### ISSUE TITLE
Decimals lose precision (postgres)

### ISSUE THREAD

**Issue by aboutlo** [prisma/prisma#6852]
## Bug description
The decimals sent to Postgres by the query engine lose precision

## How to reproduce

run this script:

```ts
import { PrismaClient } from '@prisma/client'
import { Decimal } from '@prisma/client/runtime'
const prisma = new PrismaClient()

const run = async () => {
  const value = new Decimal('600.780000000012345678912345678912')
  console.log('before', { foo: value })
  const foo = await prisma.foo.create({
    data: {
      amountDecimal: value,
      amountString: value.toString(),
    },
  })
  console.log('after', { foo })
}
run()

```

## Expected behaviour

I expect  `amountDecimal`: `600.7800000006123456789` not `600.7800000006123`,

## Prisma information
```
model Foo {
  id                 String      @id @default(uuid())
  amountDecimal      Decimal  @db.Decimal(65, 30)
  amountString       String
}
```

SQL Table generated:
```
CREATE TABLE "Foo" (
    "id" TEXT NOT NULL,
    "amountDecimal" DECIMAL(65,30) NOT NULL,
    "amountString" TEXT NOT NULL,

    PRIMARY KEY ("id")
);
```

## Environment & setup

```
prisma               : 2.21.2
@prisma/client       : 2.21.2
Current platform     : darwin
Query Engine         : query-engine e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules/@prisma/engines/query-engine-darwin)
Migration Engine     : migration-engine-cli e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine : introspection-core e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary        : prisma-fmt e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash : e421996c87d5f3c8f7eeadd502d4ad402c89464d
Studio               : 0.371.0
```

**Comment by pimeys** [comment]
Hey, sorry for slow answer (was a long day), and I'm not having a work computer in my hands anymore. It might be in `decimal.js` the issue, I've seen similar things. You can also use strings instead, so would you try that instead and tell me if you still see problems?

**Comment by aboutlo** [comment]
With my current implementation, the output is:
```
before { foo: 600.780000000012345678912345678912 }
after {
  foo: {
    id: '8cb2476d-24cb-4fae-aa08-7b493e4f78ce',
    amountDecimal: 600.7800000000124,
    amountString: '600.780000000012345678912345678912'
  }
}
```

I then changed as you suggested to string my code:
```
const foo = await prisma.foo.create({
    data: {
      amountDecimal: value.toString(),
      amountString: value.toString(),
    },
  })
```

and then  the output is
```
before { foo: 600.780000000012345678912345678912 }
after {
  foo: {
    id: '7e0e2b04-69d6-42cc-a97f-236f89545259',
    amountDecimal: 600.780000000012345678912345678912,
    amountString: '600.780000000012345678912345678912'
  }
}
```

I was expecting a TS compiler error but looks like that that `Prisma.Decimal` accepts the `Value` as a string as well.

**Comment by pantharshit00** [comment]
Hmm so indeed looks like a decimal js bug. 

I can reproduce this. We should investigate more here.

**Comment by matthewmueller** [comment]
We'll be tackling this in #6902

**Comment by SevInf** [comment]
This case is fixed by #15157
Other decimal precision issues are tracked in #8160.

**Comment by janpio** [comment]
For internal bookkeeping: This was considered a duplicate of https://github.com/prisma/prisma/issues/5925
